Source/Core/VideoCommon/LightingShaderGen.h

```diff
@@ -62,7 +62,8 @@ static void GenerateLightShader(T& object, LightingUidData& uid_data, int index,
 		case LIGHTATTN_NONE:
 		case LIGHTATTN_DIR:
 			object.Write("ldir = normalize(" LIGHT_POS".xyz - pos.xyz);\n", LIGHT_POS_PARAMS(index));
-			object.Write("attn = 1.0f;\n");
+			object.Write("attn = 1.0;\n");
+			object.Write("if (length(ldir) == 0.0)\n\t ldir = float3(1.0, 1.0, 1.0);\n");
```

Comment:
The Microsoft HLSL docs for normalize() say "If the length of the x parameter is 0, the result is indefinite".  Maybe you should perform this check differently?

Also, "float3(1.0, 1.0, 1.0)" isn't normalized, and seems like sort of a random choice.